src/plugins/kibana/public/settings/sections/indices/add_data_steps/pipeline_setup/directives/processor_ui_convert.js

```diff
@@ -27,16 +27,20 @@ app.directive('processorUiConvert', function () {
         pipeline.setDirty();
       }
 
-      $scope.types = ['integer', 'float', 'string', 'boolean'];
+      $scope.types = ['auto', 'number', 'string', 'boolean'];
 
       $scope.$watch('processor.inputObject', consumeNewInputObject);
 
       $scope.$watch('processor.sourceField', () => {
+        if (_.isEmpty($scope.processor.targetField)) {
+          $scope.processor.targetField = $scope.processor.sourceField;
+        }
```

Comment:
Given that targetField is optional, let's not autofill it (and add a label saying it's optional). Current behavior is confusing because it gets autofilled on the initial selection of `field`, but then any changes to `field` leave `targetField` alone, which caused me to overwrite the first field without realizing it.